shared/src/main/java/gov/hhs/cdc/trustedintermediary/metadata/MetaDataStep.java

```diff
@@ -0,0 +1,12 @@
+package gov.hhs.cdc.trustedintermediary.metadata;
+
+/** List of steps where we log metadata events */
+public enum MetaDataStep {
+    RECEIVED_FROM_REPORT_STREAM,
+
+    ORDER_CONVERTED_TO_OML,
+
+    CONTACT_SECTION_ADDED_TO_PATIENT,
+
+    SENT_TO_REPORT_STREAM
+}
```

Comment:
These are some great steps, but they are ETOR specific and this area of our project should contain things that are generic across any usecase of our application.

So, maybe we can use inheritance to acomplish what we want?  For example, here in the shared project we have a generic parent enum class that doesn't have any specific values, but then we have a `public enum OrderMetaDataStep` in the `etor` sub-project that extends that generic parent enum class.  Maybe that would allow us to specify a specific type in `MetaDataEntry` but allow us to specify different steps for different plugins?  Or perhaps we change `MetaDataEntry#entryStep` to be just a `String` instead of an actual type if we're OK without specific type?

Reply:
My thinking for using the enum was that it would enable us to see all of the steps in order.  To me that could assist with troubleshooting as the process inevitably grows out.  I'd prefer this over Strings, especially since we have multiples of the same class, IE: LocalFileSender and ReportStreamOrderSender

Reply:
done.

Reply:
done.